docs: document usage

HEPOSHEIKKI opened the pull request:
Improved the README and added the APISpec and themes documentation in the `docs/` directory.

This is my first time writing documentation so feel free to correct any mistakes or improve the formatting.

HEPOSHEIKKI committed:
Document some API methods

Will refactor documentation to separate files later when finished

HEPOSHEIKKI committed:
Moved APISpec to a different file

HEPOSHEIKKI committed:
Create Themes.md

Add screenshot for each theme

HEPOSHEIKKI committed:
Link to Themes.md

HEPOSHEIKKI committed:
Finish documenting all the endpoints I know of

HEPOSHEIKKI committed:
Fix wording

HEPOSHEIKKI committed:
Mention the vercel deployment

HEPOSHEIKKI committed:
Fix formatting

HEPOSHEIKKI committed:
Changed the deployment url to use https

Chicken commented on docs/Themes.md:
```diff
@@ -0,0 +1,333 @@
+# Themes
+
+Themes are used to customise the colors of the card. github-readme-testaustime ships with a whopping 60 themes to choose from.
```
We could just link the upstream's theme listing <https://github.com/anuraghazra/github-readme-stats/blob/678fa1141676fc8fb75e2a322e4297323900e330/themes/README.md>

Chicken commented on docs/APISpec.md:
```diff
@@ -0,0 +1,185 @@
+All requests for a card are made to `/api/testaustime` and parameters are passed as query strings.
+
+## Endpoints
+The API provides 17 routes:
```
Calling these endpoints/routes is wrong, it's really just a single endpoint/route with query string arguments

Chicken commented on docs/APISpec.md:
```diff
@@ -0,0 +1,185 @@
+# APISpec
+
+## Usage
```
We could again link to upstream docs and only mention differences
<https://github.com/anuraghazra/github-readme-stats/tree/678fa1141676fc8fb75e2a322e4297323900e330#common-options>
<https://github.com/anuraghazra/github-readme-stats/tree/678fa1141676fc8fb75e2a322e4297323900e330#wakatime-card-exclusive-options>
This isn't really necessary though, this is fine as is.

HEPOSHEIKKI commented:
This probably closes #1 as well

Chicken commented on docs/APISpec.md:
```diff
@@ -0,0 +1,185 @@
+    icon_color
+    border_color
+
+Used to set the color of the respective element in hex format. These values will override the current theme if one is specified.
```
Apparently the `bg_color` can be a comma seperated list for gradients

HEPOSHEIKKI committed:
Fix terminology and improve coloring section

https://github.com/Testaustime/github-readme-testaustime/pull/2#discussion_r1328064071

Chicken commented on docs/Themes.md:
```diff
@@ -0,0 +1,333 @@
+<details>
+    <summary>default</summary>
+  
+![image](https://github.com/HEPOSHEIKKI/github-readme-testaustime/assets/51783206/6a726abe-5508-412a-954d-e1823dbc630f)
```
Oh yeah, these image urls are a bit weird, not sure how they will work when merged and fork is deleted.
